src/subcommand/fix.rs

```diff
@@ -0,0 +1,484 @@
+    let fossa_connection_errors = check_fossa_connection(config).await;
+    print_errors(
+        "\nErrors found while checking integrations",
+        integration_errors,
+    );
+    print_errors(
+        "\nErrors found while checking connection to FOSSA",
+        fossa_connection_errors,
+    );
+    Ok(())
+}
+
+// If there are errors, returns a string containing all of the error messages for a section.
+// Sections are things like "checking integrations" or "checking fossa connection"
+// If there are no errors, it returns None.
+#[tracing::instrument]
+fn print_errors(msg: &str, errors: Vec<Error>) {
+    if !errors.is_empty() {
+        println!("{}\n", msg.bold().red());
+        for err in errors {
+            println!("{}", err.fix_explanation());
+        }
+    }
+}
+
+/// Check that we can connect to the integrations
+/// This is currently done by running `git ls-remote <remote>` using the authentication
+/// info from the transport.
+#[tracing::instrument(skip(config))]
+async fn check_integrations(config: &Config) -> Vec<Error> {
+    let title = "\nDiagnosing connections to configured repositories\n"
+        .bold()
+        .blue()
+        .to_string();
+    println!("{}", title);
+    let integrations = config.integrations();
+    let mut errors = Vec::new();
+    for integration in integrations.iter() {
+        match check_integration(integration).await {
+            Ok(()) => {
+                println!("✅ {}", integration.remote())
+            }
+            Err(err) => {
+                println!("❌ {}", integration.remote());
+                errors.push(err);
+            }
+        }
+    }
+    errors
+}
+
+#[tracing::instrument]
+async fn check_integration(integration: &Integration) -> Result<(), Error> {
+    let Protocol::Git(transport) = integration.protocol();
+    repository::ls_remote(transport)
+        .await
+        .or_else(|err| Error::integration_error(integration.remote(), transport, err).wrap_err())?;
+    Ok(())
+}
+
+#[tracing::instrument(skip(config))]
+async fn check_fossa_connection(config: &Config) -> Vec<Error> {
+    let title = "\nDiagnosing connection to FOSSA\n"
+        .bold()
+        .blue()
+        .to_string();
+    println!("{}", title);
+    let mut errors = Vec::new();
+
+    let get_with_no_auth = check_fossa_get_with_no_auth(config).await;
+    match get_with_no_auth {
+        Ok(()) => {
+            println!("✅ check fossa API connection with no auth required");
+        }
+        Err(err) => {
+            println!("❌ check fossa API connection with no auth required");
+            errors.push(err)
+        }
+    }
+    let get_with_auth = check_fossa_get_with_auth(config).await;
+    match get_with_auth {
+        Ok(()) => {
+            println!("✅ check fossa API connection with auth required");
+        }
+        Err(err) => {
+            println!("❌ check fossa API connection with auth required");
+            errors.push(err);
+        }
+    }
```

Comment:
[nit, optional] Generally speaking it's considered more idiomatic to explicitly ignore arguments you don't care about at all:
```suggestion
    let get_with_no_auth = check_fossa_get_with_no_auth(config).await;
    match get_with_no_auth {
        Ok(_) => {
            println!("✅ check fossa API connection with no auth required");
        }
        Err(err) => {
            println!("❌ check fossa API connection with no auth required");
            errors.push(err)
        }
    }
    let get_with_auth = check_fossa_get_with_auth(config).await;
    match get_with_auth {
        Ok(_) => {
            println!("✅ check fossa API connection with auth required");
        }
        Err(err) => {
            println!("❌ check fossa API connection with auth required");
            errors.push(err);
        }
    }
```